actors/market/tests/market_actor_test.rs

```diff
@@ -804,16 +790,26 @@ fn provider_and_client_addresses_are_resolved_before_persisting_state_and_sent_t
     deal.verified_deal = true;
 
     // add funds for client using its BLS address -> will be resolved and persisted
-    add_participant_funds(&mut rt, client_bls, deal.client_balance_requirement());
-    assert_eq!(
-        deal.client_balance_requirement(),
-        get_escrow_balance(&rt, &client_resolved).unwrap()
-    );
+    let amount = deal.client_balance_requirement();
+
+    rt.set_value(amount.clone());
+    rt.set_caller(*ACCOUNT_ACTOR_CODE_ID, client_resolved);
+    rt.expect_validate_caller_any();
+    assert!(rt
+        .call::<MarketActor>(
```

Comment:
why go out of our way to not use `add_participant_funds`?

Reply:
Because `add_participant_funds` uses `rt.set_caller`, which _should_ only be called with ID addresses (since the `caller()` method is expected to always return ID).

The change to "fix" this `set_caller` method was made as part of this changeset, and so this test (which specifically requires the `caller` and the parameter address to be of different types) needed to change.